### PR TITLE
Task8

### PR DESCRIPTION
**Pull request by iharyachnik**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -8,6 +8,21 @@ import './auth';
 
 const app = express();
 
+var allowCrossDomain = function(req, res, next) {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE');
+  res.header('Access-Control-Allow-Headers', 'Content-Type, Authorization');
+    
+  // intercept OPTIONS method
+  if ('OPTIONS' == req.method) {
```
Comment (DraXu3) [inline app.js]: `==` - really?

File: ngmp/app.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var SwaggerExpress = require('swagger-express-mw');
+var app = require('express')();
+module.exports = app; // for testing
```
Comment (DraXu3) [inline ngmp/app.js:5]: Why not to export it for testing?

File: ngmp/app.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+SwaggerExpress.create(config, function(err, swaggerExpress) {
+  if (err) { throw err; }
+
+  // install middleware
+  swaggerExpress.register(app);
+
+  var port = process.env.PORT || 10010;
+  app.listen(port);
+
+  if (swaggerExpress.runner.swagger.paths['/hello']) {
```
Comment (DraXu3) [inline ngmp/app.js:20]: What is it for?

File: ngmp/config/default.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# swagger configuration file
```
Comment (DraXu3) [inline ngmp/config/default.yaml:1]: Was this file auto generated?

File: .vscode/launch.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (DraXu3) [inline .vscode/launch.json]: What this is for?

**Review by DraXu3 — COMMENTED** [review]
Please resolve all remarks provided by comments

File: ngmp/app.js (hunk 3 above)
Comment (iharyachnik) [reply ngmp/app.js:20]: This middleware is required for options requests due to CORS . In these requests authentication should not been provided. That's why I added it  https://gist.github.com/cuppster/2344435

**Review by DraXu3 — APPROVED** [review]

**Pull request merged**